### PR TITLE
feat(providers): first implementation of a 1password connect provider

### LINKED ISSUE #54: feat: support 1password Secrets Automation
With the recent acquisition of SecretHub and a released feature called Secrets Automation, 1Password became an interesting solution as a secret manager for the whole infrastructure.

https://1password.com/secrets/
https://support.1password.com/secrets-automation/
https://github.com/1Password/connect-sdk-go

It would be great to have it as a possible backend.

### PR DESCRIPTION
**Pull request by juckerf**
This PR adds a provider for 1password connect (https://developer.1password.com/docs/connect).
From the `README`:

> For this provider to work you require a working and accessible [1Password connect server](https://developer.1password.com/docs/connect).
> The following env vars have to be configured:
> - `OP_CONNECT_HOST`
> - `OP_CONNET_TOKEN`
> 
> 1Password is organized in vaults and items.
> An item can have multiple fields with or without a section. Labels can be set on fields and sections.
> Vaults, items, sections and labels can be accessed by ID or by label/name (and IDs and labels can be mixed and matched in one URL).
> 
> If a section does not have a label the field is only accessible via the section ID. This does not hold true for some default fields which may have no section at all (e.g.username and password for a `Login` item).
> 
> *Caution: vals-expressions are parsed as URIs. For the 1Password connect provider the host component of the URI identifies the vault (by ID or name). Therefore vaults containing certain characters not allowed in the host component (e.g. whitespaces, see [RFC-3986](https://www.rfc-editor.org/rfc/rfc3986#section-3.2.2) for details) can only be accessed by ID.*
> 
> Examples:
> 
> - `ref+onepasswordconnect://VAULT_ID/ITEM_ID#/[SECTION_ID.]FIELD_ID`
> - `ref+onepasswordconnect://VAULT_LABEL/ITEM_LABEL#/[SECTION_LABEL.]FIELD_LABEL`
> - `ref+onepasswordconnect://VAULT_LABEL/ITEM_ID#/[SECTION_LABEL.]FIELD_ID`

If merged, the PR resolves #54 - but there is one topic we should discuss first.
Using URIs **without a fragment** is most probably useless in the current implementation. To support any combination of labels and IDs in the URI, the string map is populated with every possible combination of label and ID for its keys which does not lead to meaningful data.
Currently I have no proper solution for that. Maybe 1password is not suited for this use case? (then we should simply drop the `stringprovider` ability) Or we should introduce `PARAMS`/query components to control the behavior of the `stringprovider`? (like `preferLabels` to use labels for the map whenever possible sacrificing the ability to match keys by ID?)
Any ideas are appreciated. And since I'm not really fluent in go I'm also open to feedback regarding codestyle, stability etc :-)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -627,6 +629,27 @@ Examples:
 - `ref+gitlab://gitlab.com/11111/password`
 - `ref+gitlab://my-gitlab.org/11111/password?ssl_verify=true&scheme=https`
 
+### 1Password Connect
+
+For this provider to work you require a working and accessible [1Password connect server](https://developer.1password.com/docs/connect).
+The following env vars have to be configured:
+- `OP_CONNECT_HOST`
```
Comment (yxxhero) [inline README.md:636]: what's the meaning of OP? maybe it's better to use the full name.

Comment (juckerf) [reply README.md:636]: These are the default env variables the 1Password connect SDK uses (I guess it's an abbreviation for  "OnePassword")
But if the naming is an issue I'm happy to define custom environment variables (e.g. `ONEPASSWORDCONNECT_HOST` and `ONEPASSWORDCONNECT_TOKEN`) and use them to create the 1password connect API client.

Comment (yxxhero) [reply README.md:636]: got it. no need to change.

File: pkg/providers/onepasswordconnect/onepasswordconnect.go [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+	return p
+}
+
+// Get secret string from 1Password Connect
+func (p *provider) GetString(key string) (string, error) {
+	var err error
+
+	splits := strings.Split(key, "/")
+	if len(splits) < 2 {
+		return "", fmt.Errorf("invalid URI: %v", errors.New("vault or item missing"))
```
Comment (yxxhero) [inline pkg/providers/onepasswordconnect/onepasswordconnect.go:31]: vault?

**Comment by juckerf** [comment]
1password is organized in vaults -> items -> [sections ->] fieldsSee also description of the vals-URI for this provider in the READMEOn 21 Sep 2023 15:07, yxxhero ***@***.***> wrote:
@yxxhero commented on this pull request.



In pkg/providers/onepasswordconnect/onepasswordconnect.go:
> +}
+
+// New creates a new 1Password Connect provider
+func New(cfg api.StaticConfig) *provider {
+	p := &provider{}
+
+	return p
+}
+
+// Get secret string from 1Password Connect
+func (p *provider) GetString(key string) (string, error) {
+	var err error
+
+	splits := strings.Split(key, "/")
+	if len(splits) < 2 {
+		return "", fmt.Errorf("invalid URI: %v", errors.New("vault or item missing"))

vault?

File: pkg/providers/onepasswordconnect/onepasswordconnect.go (hunk 2 above)
Comment (juckerf) [reply pkg/providers/onepasswordconnect/onepasswordconnect.go:31]: 1password is organized in vaults -> items -> [sections ->] fields
See also description of the vals-URI for this provider in the README

**Comment by yxxhero** [comment]
@juckerf BTW. could you add some tests for this feature? As the first implementation of a 1password connect provider. it's good for me.

**Comment by juckerf** [comment]
thanks for the merge!

**Pull request merged**